File: Solutions/1342. Number of Steps to Reduce a Number to Zero/CPP/Number_of_Steps_to_Reduce_a_Number_to_Zero.cpp
#include <iostream>

using namespace std;

class Solution {
public:
    int numberOfSteps(int num) {
        if (num == 0) {
            return 0;
        }

        int numOf[2] = {0};
        while (num) {
            int bit    = num & 1;
            numOf[bit] = numOf[bit] + 1;
            num        = num >> 1;
        }

        return numOf[0] + numOf[1] * 2 - 1;
    }
};

int main() {
    Solution S;

    // input
    int num = 14;

    cout << S.numberOfSteps(num) << endl;

    return 0;
}

File: Solutions/1342. Number of Steps to Reduce a Number to Zero/Python/Number_of_Steps_to_Reduce_a_Number_to_Zero.py
class Solution:
    def numberOfSteps(self, num: int) -> int:
        if num == 0:
            return 0
        else:
            return format(num, 'b').count('0') + format(num, 'b').count('1') * 2 - 1

S = Solution()

print(S.numberOfSteps(14))
